src/app/_components/Event/EventFilter.js

```diff
@@ -0,0 +1,179 @@
+'use client';
+
+import { useState, useEffect } from 'react';
+import { useInfiniteQuery } from '@tanstack/react-query';
+import { useInView } from 'react-intersection-observer';
+import { useSearchParams } from 'next/navigation';
+import { CircularProgress } from '@mui/material';
+import CheckMark from '@icons/check-mark.svg';
+import Search from '@icons/search.svg';
+import { allEvents, useEventSetParam } from '@hooks';
+import { cn } from '@/utils/cn';
+import { PillButton } from '../PillButton';
+import { buttonColorVariant } from '../PillButton/style';
+import { NoInfoToShow } from '../NoInfoToShow';
+import { EventCard } from './Card';
+import { monthsAhead, monthNames } from './config';
+
+export function EventFilter() {
+  const [months, setMonths] = useState([]);
+  const [activeIndex, setActiveIndex] = useState(0);
+  const [activeMonth, setActiveMonth] = useState(months[0]);
+  const [activeMonthNumber, setActiveMonthNumber] = useState('');
+  const [hoveredIndexes, setHoveredIndexes] = useState(new Array(months.length));
+
+  const currentMonth = new Date().getMonth() + 1;
+
+  const searchParams = useSearchParams();
+  const { addParam, deleteParam } = useEventSetParam('month');
+
+  const { ref, inView } = useInView();
+
+  const { semiorange } = buttonColorVariant.eventFilter;
+
+  const activeButtonStyles = cn({
+    'pointer-events-none border-secondary-300 bg-secondary-300 font-semibold text-gray-900': semiorange.active,
+  });
+
+  function selectButtonIcon(index) {
+    if (activeIndex === index && semiorange.active) {
+      return <CheckMark key={`checkmark+${index}`} className="h-4 w-4 transition-all" />;
+    }
+    if (hoveredIndexes[index] && activeIndex !== index && (semiorange.hover || semiorange.focused)) {
+      return <Search key={`searchicon+${index}`} className="h-4 w-4 transition-all " />;
```

Comment:
@Yivgha Let's always render Search icon for buttons that are not === activeIndex, but hide it with `hidden` by default. Then we can make it `group-hover:block` when user hover over button.